lib/jsonapi_parameters/validator.rb

```diff
@@ -0,0 +1,52 @@
+require 'active_model'
+require 'json_schemer'
+
+module JsonApi::Parameters
+  SCHEMA_PATH = 'spec/support/jsonapi_schema.json'.freeze
+
+  private
+
+  def should_prevalidate?
+    JsonApi::Parameters.enforce_prevalidation && !JsonApi::Parameters.suppress_validation_errors
+  end
+
+  class Validator
```

Comment:
I would call this JsonApiDotOrgValidator or something like that

Reply:
I don't like long names. They remind me of Java, and Ruby is not Java. Why would you call it this way?

Reply:
Because of schema this validator handles. Also "Validator" is not descriptive in a way that would suggest what it is using as basis for Validation

Reply:
There might be multiple flavours of schema

Reply:
In our case there's only one flavour of schema, as this library only targets JsonApi. I agree though that indeed it might be unclear when we instantiate it by calling `Validator` directly, and I think @pstrzalk is on the good path here. 
I'm changing it to use the full namespace in every call.